### PR TITLE
Add support for custom @Nullable annotations

### LINKED ISSUE #99: Support @CheckForNull annotation
```
error: [NullAway] method returns @Nullable, but superclass method com.github.benmanes.caffeine.cache.AsyncLoadingCache.getIfPresent(java.lang.Object) returns @NonNull
  public @Nullable CompletableFuture<V> getIfPresent(@Nonnull Object key) {
```

However, this method is annotated with `@CheckForNull` which should implicitly mean `@Nullable`.

### PR DESCRIPTION
**Pull request by lazaroclapp**
And @CheckForNull as a default Nullable annotation, together with
@Nullable and @NullableDecl.

Includes unit tests, but might need a manual performance regression test before landing.

Fixes #99

### CODE REVIEW

**Comment by msridhar** [comment]
I want to try benchmarking this change before approving.  Will try to find time soon.

File: nullaway/src/main/java/com/uber/nullaway/Nullness.java [hunk 1]
```diff
@@ -127,26 +128,41 @@ public String toString() {
...
-   */
-  public static boolean isNullableAnnotation(String annotName) {
-    return annotName.endsWith(".Nullable")
-        || annotName.equals("org.checkerframework.checker.nullness.compatqual.NullableDecl");
-  }
+    /**
+     * @param annotName annotation name
+     * @return true if we treat annotName as a <code>@Nullable</code> annotation, false otherwise
+     */
+    public boolean isNullableAnnotation(String annotName) {
```
Comment (kageiit) [inline nullaway/src/main/java/com/uber/nullaway/Nullness.java]: you could potentially memoize the result of this as well with a reasonably sized lru cache so you can avoid looking up the same annotation multiple times

Comment (lazaroclapp) [reply nullaway/src/main/java/com/uber/nullaway/Nullness.java]: How much faster do we expect the dictionary lookup to be compared to iterating over a set of `n~=5`? I don't think there will ever be a huge set of `@Nullable` equivalent annotations here. That said, we could benchmark both approaches, for sure!

Comment (kageiit) [reply nullaway/src/main/java/com/uber/nullaway/Nullness.java]: We are looking at all instances of annotations used in the source code

Everytime you lookup an annotation, you keep a bit about whether it is nullable or not.

The source code can refer to the same annotation in many places so it saves us some time not needing to compare against an ever growing list of nullable annotations

**Comment by lazaroclapp** [comment]
This is frankly so out of sync with master at this point that re-doing the PR would be easier than updating, and it's not clear we'd even want to, based on the discussion above. Closing.

**Pull request closed without merging**